screens/ProductMainScreen.js

```diff
@@ -19,6 +20,9 @@ export default function ProductMainScreen({ navigation }) {
     showRefineModal();
   };
 
+  const toMapView = () => {
+    navigation.navigate('Map');
+  };
```

Comment:
If you navigate to map screen, it will override the buttonTabNavigator. My original suggestion meant that if user clicks on 'map button' it will update the state and toggle the view. It will not have a separate map screen. 
 
For example, in ProductMainScreen.js ...
`
 const [mapView, setMapViewVisibility] = useState(false);

return (
<SafeAreaView style={[t.flex1, t.bgWhite]}>
{!mapView ? (<View>
// List View
// If user click on the map button => setMapViewVisibility(true)
</View>) : (<View>
// Map View
// If user click on the map button => setMapViewVisibility(false)
</View>)}
</SafeAreaView>


Let me know if that makes sense for you? I am definitely open to any alternative option. 


`